Archive/Imo/Imo2024Q1.lean

```diff
@@ -68,31 +70,31 @@ lemma mem_Ico_one_of_mem_Ioo (h : α ∈ Set.Ioo 0 2) : α ∈ Set.Ico 1 2 := by
   have hr : 1 < ⌈α⁻¹⌉₊ := by
     rw [Nat.lt_ceil]
     exact_mod_cast one_lt_inv h0 hn
-  replace hc := hc ⌈α⁻¹⌉₊ (zero_lt_one.trans hr)
-  refine hr.ne' ?_
+  apply hr.ne'
   suffices ⌈α⁻¹⌉₊ = (1 : ℤ) from mod_cast this
-  refine Int.eq_one_of_dvd_one (Int.zero_le_ofNat _) ?_
-  convert hc
+  apply Int.eq_one_of_dvd_one (Int.zero_le_ofNat _)
+  convert hc ⌈α⁻¹⌉₊ (zero_lt_one.trans hr)
   rw [← Finset.add_sum_Ico_eq_sum_Icc hr.le]
   convert (add_zero _).symm
   · rw [Int.floor_eq_iff]
-    refine ⟨?_, ?_⟩
+    constructor
     · rw [Int.cast_one]
       calc 1 ≤ α⁻¹ * α := by simp [h0.ne']
-        _ ≤ ⌈α⁻¹⌉₊ * α := by gcongr; exact Nat.le_ceil _
-    · calc ⌈α⁻¹⌉₊ * α < (α⁻¹ + 1) * α := by gcongr; exact Nat.ceil_lt_add_one (inv_nonneg.2 h0.le)
+        _ ≤ ⌈α⁻¹⌉₊ * α := (mul_le_mul_right h0).2 (Nat.le_ceil α⁻¹)
+    · calc ⌈α⁻¹⌉₊ * α
+        _ < (α⁻¹ + 1) * α := (mul_lt_mul_right h0).2 (Nat.ceil_lt_add_one (inv_nonneg.2 h0.le))
         _ = 1 + α := by field_simp [h0.ne']
-        _ ≤ (1 : ℕ) + 1 := by gcongr; norm_cast
-  · refine Finset.sum_eq_zero ?_
+        _ ≤ (1 : ℕ) + 1 := add_le_add Nat.cast_one.symm.le hn.le
```

Comment:
I personally prefer the `gcongr; norm_cast` version.

Reply:
Only here, or everywhere?

Reply:
Is there generally any preference in imo files to use tactic- or term-mode?

Maybe

``` ... := by
  gcongr
  norm_cast
  apply ...
```

in that case?

Reply:
Undo this change, or all changes to terms in `calc` ?

Reply:
I would prefer to revert all of them. `gcongr` is a very natural atomic step in my opinion. I think we should showcase it.
But my opinion isn't very strong.

Reply:
All right, I will revert them all.